152_data_reading_program/sensor_reader.py

```diff
@@ -2,50 +2,81 @@
 # in case it is not already set.
 # If this system call is not made, import board will fail
+# This system call sometimes works on my computer but usually fails
 os.system("export BLINKA_MCP2221=1")
-import time # For sleep function
-import adafruit_scd30 # Adafruit's Library for the SCD-30
-import busio # For I2C
+
+try:
+   os.environ["BLINKA_MCP2221"]
+except KeyError:
+   print(f'Before running this program, type "export BLINKA_MCP2221=1"'
+         f'in the same terminal window.')
+   sys.exit()
+
 import board # For MCP-2221
 
+# Variables to enhance precision of sensor
+
+# Temperature Offset - Ideally you should have a temperature sensor that is
+# independent of this sensor since this sensor generates heat as it is used
+# The temperature offset is the difference between ambient temperature and
+# the onboard temperature sensor. T_offset = Tscd_30 - Treferefence
+# set with scd.temperature_offset = OFFSET_VALUE. Every tick is 0.01*C (uint16)
+# Uses last value set before repowering.
+
+# There are also methods for calibration of the sensor but since they come calibrated
+# it might not be good to change unless you have a sensor to compare it against or
+# have the right environment for autocalibration (requires access to fresh air, and 5 days).
+
+
 # From Carter's Script, it is recommended to start with a frequency
-# of 50 Khz due to "Clock Stretching.
+# of 50 Khz due to "Clock Stretching".
 i2c = busio.I2C(board.SCL, board.SDA, frequency=50000)
 scd = adafruit_scd30.SCD30(i2c)
 
-# scd.reset() apparently isn't necessary as long as the sensor is not in an
-# error state to start with. scd.reset() is alternative to the
-# following which is supposed to work better for MCP-2221
-def reset_scd():
-    #Check if this script is run on MCP-2221 (Carter's )
-    if hasattr(i2c, "_i2c"):
-        if hasattr(i2c._i2c, "_mcp2221"):
-            # Run the MCP-2221 reset command
-            i2c._i2c._mcp2221._reset()
-    else: # Run the normal reset command (for Rasperry pi GPIO for instance)
-        scd.reset()
+# The scd.reset() method is known to cause issues with the MCP-2221
+# A script written by Adafruit offered an alternative reset method
+# but it also caused issues after testing, so it is best never to
+# invoke the reset method.
+
+
+# Alitude, set with with scd.altitude = ALTITUDE (uint16, height above sea level in meters)
+# Seting altitude is disregarded when ambient pressure is given to the system.
+# Ambient Pressure, set with scd.ambient_pressure = PRESSURE_VALUE. Default
+# value is 1013.25 mBar   uint16 700 to 1400 range, 0 sets to default.
+def get_altitude():
+   if len(sys.argv) > 1:
+      if sys.argv[1].isnumeric():
+         altitude = int(sys.argv[1])
+         print(f"Altitude entered: {altitude} m")
+         scd.altitude = altitude
+      else:
+         print("Parameter must be an int for altitude, defaulting to 1013.25 mBar")
+   else:
+      print("Defaulting to 1013.25 mBar")
```

Comment:
A fancier way to handle command line arguments is with `argparse`.  If you need something quick, `sys.argv` is fine.

Instead of using an `if`/`else` with `.isnumeric()`, better to use a `try`/`except` on the `int()`:
```py
try:
    altitude = int(sys.argv[1])
except ValueError:
    print("Argument for altitude must be an int.  Defaulting to 1013.25 mBar")
else:
    print(f"Altitude entered: {altitude} m")
    scd.altitude = altitude
```
Also this block is indented with 3 spaces instead of 4.